src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -943,43 +970,100 @@ protected override void ValidateElement(object element)
                         Metadata.ValidateNotNullFailure);
             }
 
-            var o = element as PSObject;
-            if (o != null)
+            if(_rangeKind.HasValue)
             {
-                element = o.BaseObject;
+                // Using dynamic to handle different type of numbers such as
+                // int, double, decimal and float. 
+                dynamic val = element;
```

Comment:
Reusing ```GetCommonType``` looked promising but failed in certain scenarios. 

For example, if the incoming element was ```-1```, this value was interpreted as a string. This would cause the ```GetCommonType``` method to fail since it could not find the common group between ```0``` and the string.

I've resorted to using the ```LanguagePrimitives.TryConvertTo``` method and casting the incoming value as a double since this is the largest number type currently supported by the function. 

Is there a cleaner way to test if the object is comparable with zero that I might be missing?